PowerPointLabs/Test/FunctionalTest/TimerLabTest.cs

```diff
@@ -174,6 +212,145 @@ private void TestNonMultipleDenominationDurationWithCountdownTimer(ITimerLabCont
             AssertIsSame(OriginalSlideNo, CountdownAndNonMultipleDenominationDurationSlideNo);
         }
 
+        private void TestEditProgressBarState(ITimerLabController timerLab)
+        {
+            PpOperations.SelectSlide(OriginalSlideNo);
+            timerLab.SetProgressBarCheckBoxState(true);
+
+            AssertIsSame(OriginalSlideNo, ProgressBarCheckedSlideNo);
+        }
+
+        private void TestCreateInitialTimerPb(ITimerLabController timerLab)
+        {
+            PpOperations.SelectSlide(PbOriginalSlideNo);
+            timerLab.SetProgressBarCheckBoxState(true);
+            timerLab.SetDurationTextBoxValue(1.05);
+            timerLab.ClickCreateButton();
+            AssertIsSame(PbOriginalSlideNo, PbInitialTimerSlideNo);
+        }
+
+        private void TestEditTimerWidthPb(ITimerLabController timerLab)
+        {
+            PpOperations.SelectSlide(PbOriginalSlideNo);
+            timerLab.SetWidthTextBoxValue(250);
+            AssertIsSame(PbOriginalSlideNo, PbChangeWidthSlideNo);
+        }
+
+        private void TestEditTimerHeightPb(ITimerLabController timerLab)
+        {
+            PpOperations.SelectSlide(PbOriginalSlideNo);
+            timerLab.SetHeightTextBoxValue(400);
+            AssertIsSame(PbOriginalSlideNo, PbChangeHeightSlideNo);
+        }
+
+        private void TestDeleteTimerBodyPb(ITimerLabController timerLab)
+        {
+            PpOperations.SelectSlide(PbOriginalSlideNo);
+            ShapeRange shapes = PpOperations.SelectShape(TimerBody);
+            shapes.Delete();
+
+            MessageBoxUtil.ExpectMessageBoxWillPopUp("Error",
+               "Only one timer allowed per slide.", timerLab.ClickCreateButton, "Ok");
+            AssertIsSame(PbOriginalSlideNo, PbRecreateBodySlideNo);
+        }
+
+        private void TestEditLineColorAndDeleteTimeMarkerPb(ITimerLabController timerLab)
+        {
+            PpOperations.SelectSlide(ChangeLineColorAndRecreateTimeMarkerSlideNo);
+            int expectedColor = PpOperations.SelectShape(TimerLineMarkerGroup)[1].Line.ForeColor.RGB;
```

Comment:
It seems this block of code has reuse value. There are some repetitions. I think it would be good to extract the code below into a reusable function, with a comment of what it is for.

```
            int expectedColor = PpOperations.SelectShape(TimerLineMarkerGroup)[1].Line.ForeColor.RGB;

            PpOperations.SelectSlide(PbOriginalSlideNo);
            ShapeRange lineMarkerGroup = PpOperations.SelectShape(TimerLineMarkerGroup);
            lineMarkerGroup.Line.ForeColor.RGB = expectedColor;
            ShapeRange timeMarkerGroup = PpOperations.SelectShape(TimerTimeMarkerGroup);
            timeMarkerGroup.Delete();
```

Reply:
I think it is not very meaningful to abstract these functions. The level of abstraction seems right as in other functions as wel.